_posts/2022-05-18-My-OTP-25-highlights.md

```diff
@@ -0,0 +1,713 @@
+
+# New option `short` for `erlang:float_to_list/2` and `erlang:float_to_binary/2` 
+
+A new option called `short` has been added to the functions `erlang:float_to_list/2` and `erlang:float_to_binary/2`. This option creates the shortest correctly rounded string representation of the given float that can be converted back to the same float again.
+
+If option `short` is specified, the float is formatted
+with the smallest number of digits that still guarantees that
+```erlang
+F =:= list_to_float(float_to_list(F, [short]))
+``` 
+When the float is inside the range (-2⁵³, 2⁵³), the notation
+that yields the smallest number of characters is used (scientific
+notation or normal decimal notation). Floats outside the range
+(-2⁵³, 2⁵³) are always formatted using scientific notation to avoid confusing 
+results when doing arithmetic operations.
+
+The implementation is contributed by Thomas Depierre and uses the Ryu algorithm.
+
+We present Ryū, a new routine to convert binary floating point numbers to their decimal representations using only fixed-size integer operations, and prove its correctness. Ryū is simpler and approximately three times faster than the previously fastest implementation.
+https://github.com/ulfjack/ryu
+
+
+# The new module `peer` supersedes the slave module
+
+ The `peer` module provides functions for starting linked Erlang nodes. The Erlang node spawning new "peer" nodes is called `origin`, and the newly started nodes are peers. 
+ 
+ A peer node automatically terminates when it loses the control connection to the origin. This connection could be an Erlang distribution connection, or an alternative - TCP or standard I/O. The alternative connection provides a way to execute remote procedure calls even when Erlang Distribution is not available, allowing to test the distribution itself.
+
+Peer node terminal input/output is relayed through the origin. If a standard I/O alternative connection is requested, console output also goes via the origin, allowing debugging of node startup and boot script execution (see -init_debug). File I/O is not redirected, contrary to slave(3) behaviour.
+
+The peer node can start on the same or a different host (via ssh) or in a separate container (for example Docker). When the peer starts on the same host as the origin, it inherits the current directory and environment variables from the origin.
+
+## Note
+
+This module is designed to facilitate multi-node testing with Common Test. Use the ?CT_PEER() macro to start a linked peer node according to Common Test conventions: crash dumps written to specific location, node name prefixed with module name, calling function, and origin OS process ID). Use `random_name/1` to create sufficiently unique node names if you need more control.
+
+A peer node started without alternative connection behaves similarly to `slave(3)`. When an alternative connection is requested, the behavior is similar to 
+`test_server:start_node(Name, peer, Args)`. 
```

Comment:
Do we really want to mention `test_server:start_node`? It is an internal API that should(?) only have been used in our own tests.